### PR TITLE
Save favourite artists

### PR DESCRIPTION
**Pull request by CarlaLeal**
This PR is related to the story "User can view details related to favourited artists"
**Tasks**
Adjust game summary component so users can save artists. 
Query flask API to store saved artists
Display list of artists and article details 

**Testing**
- Do you see a list of artists from the game?
- Can you save an artist from the game?
- Can you unsave an artist from the game?
- When you go to profile, can you see the artists you saved and are the ones you unsaved omitted?

### CODE REVIEW

File: frontend/frontend/src/components/Home copy.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<template>
```
Comment (abcotter) [inline frontend/frontend/src/components/Home copy.vue]: What is this copy used for? if it isn't needed for functionality please remove :)

File: frontend/frontend/src/components/SavedArtist.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+				class="saveButton"
+				v-on:click="
+					savedArtist
+						? updateArtistRelationship('remove')
+						: updateArtistRelationship('save')
+				"
+			>
+				{{ savedArtist ? "UNSAVE" : "SAVE" }}
+			</button>
+		</div>
```
Comment (abcotter) [inline frontend/frontend/src/components/SavedArtist.vue]:
```suggestion
<div class="artist-bin">
		<div class="left">
			<p>{{ artistName }}</p>
		</div>
		<div class="right">
			<button
				class="saveButton"
				v-on:click="
					savedArtist
						? updateArtistRelationship('remove')
						: updateArtistRelationship('save')
				"
			>
				{{ savedArtist ? "UNFAVOURITE" : "FAVOUTIRE" }}
			</button>
		</div>
</div>
```

Comment (abcotter) [reply frontend/frontend/src/components/SavedArtist.vue]: I'm making some changes here, to allow for better styling in the css

File: frontend/frontend/src/components/SavedArtist.vue [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+			axios.post(baseURI, postData).then(() => {
+				this.savedArtist = this.savedArtist ? false : true;
+			});
+		},
+	},
+};
+</script>
+
+<style scoped>
+.left {
```
Comment (abcotter) [inline frontend/frontend/src/components/SavedArtist.vue:70]:
```suggestion
.artists {
	display: flex;
	margin: auto;
	justify-content: space-around;
}
.artist-bin {
	display: flex;
	justify-content: space-between;
	width: 300px;
}
.left {
```

File: frontend/frontend/src/components/SavedArtist.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	display: inline-block;
+	padding-left: 10px;
+}
```
Comment (abcotter) [inline frontend/frontend/src/components/SavedArtist.vue:73]:
```suggestion
}
.right {
	display: inline-block;
	padding-left: 10px;
	padding-top: 12px;
}
```

File: frontend/frontend/src/components/SavedArtist.vue [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+.saveButton {
+	/* button */
+	width: 55px;
```
Comment (abcotter) [inline frontend/frontend/src/components/SavedArtist.vue]:
```suggestion
	margin-top: 2px;
	width: 95px;
```

File: server/app.py [hunk 6]
```diff
@@ -157,6 +157,53 @@ def updateTime():
...
+        if not artistSaved(userID, artistName):
+            saveArtist(userID, artistName)
+        return jsonify('Success'), 200
+    else:
+        unsaveArtist(userID, artistName)
+        return jsonify('Success'), 200
+
+def artistSaved(userID, artistName):
+    with DbSelector() as d:
+        query = "SELECT * FROM  UserArtist WHERE userID=%s AND artist=%s"
```
Comment (abcotter) [inline server/app.py]: some of your queries have UserID, but our DB uses UserId... I never had any issues... but maybe for consistency make sure the queries match

File: frontend/frontend/src/components/Profile.vue [hunk 7]
```diff
@@ -0,0 +1,47 @@
+<template>
+	<div>
+		<h1>Profile</h1>
```
Comment (abcotter) [inline frontend/frontend/src/components/Profile.vue:4]:
```suggestion
		<button v-on:click="routeToHome" class="back-button">BACK</button>
		<h1>Profile</h1>
```

Let's add a back button! so the user can always route home

File: frontend/frontend/src/components/Profile.vue [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+		return {
+			artists: [],
+		};
+	},
+	mounted() {
+		const baseURI = process.env.VUE_APP_HOST_URL + "api/artist/saved";
+		const postData = { userID: this.$store.state.userId };
+		axios.post(baseURI, postData).then((response) => {
+			this.artists = response.data;
+		});
```
Comment (abcotter) [inline frontend/frontend/src/components/Profile.vue]:
```suggestion
		});
	methods: {
		routeToHome: function () {
			this.$router.push("/home");
		},
	},
```

File: frontend/frontend/src/components/Profile.vue [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+	},
+};
+</script>
+
+<style scoped>
+.card {
```
Comment (abcotter) [inline frontend/frontend/src/components/Profile.vue:52]:
```suggestion
.back-button {
	position: absolute;
	left: 10px;
	top: 10px;
	/* button */
	margin-top: 2px;
	width: 95px;
	height: 23px;
	border: 2px solid #ffe381;
	background-color: #ffe381;
	/* text */
	font-size: 15px;
	font-weight: bold;
	color: #cf4985;
}
.card {
```

**Review by abcotter — CHANGES_REQUESTED** [review]
Looking good Carla, I did notice a rather long delay when routing to the profile and when the users artists load... I think either you should add a loading spinner or pulle the users favourite artists earlier and store them.

currently until the artists are loaded from the API it is displaying the message "No artists favourited yet" which isn't technically true, they are just being loaded in.

File: frontend/frontend/src/components/Home copy.vue (hunk 1 above)
Comment (CarlaLeal) [reply frontend/frontend/src/components/Home copy.vue]: No clue! Will remove :)

**Comment by CarlaLeal** [comment]
> Looking good Carla, I did notice a rather long delay when routing to the profile and when the users artists load... I think either you should add a loading spinner or pulle the users favourite artists earlier and store them.
> 
> currently until the artists are loaded from the API it is displaying the message "No artists favourited yet" which isn't technically true, they are just being loaded in.

That's a really good point! Let me investigate the loading spinner idea. I don't think the saved artists should be in a global store if they're only going to be used by one component realistically.

**Review by abcotter — COMMENTED** [review]
@CarlaLeal Looks great! There is only one issue. The "favourited" artists don't seem to be updated and accurate on the game summary page. 

I favourited an artist--> navigated home --> looked at my profile --> went home --> started another game --> then on the game summary page the artist i had previously se to favourited wasn't set correctly

File: frontend/frontend/src/components/SavedArtist.vue [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+	</div>
+</template>
+
+<script>
+import axios from "axios";
+export default {
+	props: ["artistName"],
+	data() {
+		return {
+			savedArtist: false,
```
Comment (abcotter) [inline frontend/frontend/src/components/SavedArtist.vue]: I think you should keep a local array in the vuex store called artists or something and when the person logs in we can grab the names of artists they have favourited, then update it locally when we update the database. this way we can set the savedArtist variable based on if the vuex store variable .includes() the artistName

**Review by abcotter — CHANGES_REQUESTED** [review]

**Review by abcotter — APPROVED** [review]
Looks good

**Comment by FatimahAreola** [comment]
I think we approved and merged before I completed my review. Understandable since I was having issues with my system. I went back and tested anyway, while a little slow, things are working on my end.

**Pull request merged**